tests/test_metadata_integration.py

```diff
@@ -0,0 +1,269 @@
+                for d in metadata_dir.iterdir()
+                if d.is_dir() and d.name.startswith("run_")
+            ]
+
+            assert (
+                len(run_dirs) == 2
+            ), f"Should have exactly two run directories, found: {run_dirs}"
+
+            # Verify both have model subdirectories with entity names
+            model_names = ["test_model1", "test_model2"]
+            found_models = []
+
+            for run_dir in run_dirs:
+                model_parent_dir = run_dir / "model"
+                assert (
+                    model_parent_dir.exists()
+                ), f"Model parent directory should exist in {run_dir}"
+
+                # Find which model is in this run directory
+                for model_name in model_names:
+                    model_dir = run_dir / "model" / model_name
+                    if model_dir.exists():
+                        found_models.append(model_name)
+                        break
+                else:
+                    assert False, f"No model directory found in {run_dir}"
+
+            # Verify we found both models
+            assert (
+                len(found_models) == 2
+            ), f"Should find both models, found: {found_models}"
+            assert set(found_models) == set(
+                model_names
+            ), f"Should find correct models: {model_names}, found: {found_models}"
+
+    def test_metadata_directory_permissions_and_structure(self):
+        """Test that metadata directories are created with correct permissions"""
+        with tempfile.TemporaryDirectory() as temp_dir:
+            exp = Experiment("test_metadata_perms", exp_path=temp_dir, launcher="local")
+
+            model = exp.create_model(
+                "test_model",
+                run_settings=exp.create_run_settings("echo", ["permissions"]),
+            )
+
+            exp.start(model, block=False)
+            exp.poll(interval=1)
+
+            # Check directory structure and permissions
+            smartsim_dir = pathlib.Path(temp_dir) / CONFIG.smartsim_base_dir
+            metadata_dir = smartsim_dir / "metadata"
+
+            # Verify directories exist and are readable/writable
+            assert metadata_dir.exists() and metadata_dir.is_dir()
+            assert (
+                metadata_dir.stat().st_mode & 0o700
+            )  # Owner should have read/write/execute
+
+            run_dirs = [
+                d
+                for d in metadata_dir.iterdir()
+                if d.is_dir() and d.name.startswith("run_")
+            ]
+            if run_dirs:
+                run_dir = run_dirs[0]
+                assert run_dir.exists() and run_dir.is_dir()
+
+                # Check for entity-specific model directory with entity name
+                model_dir = run_dir / "model" / "test_model"
+                if model_dir.exists():
+                    assert model_dir.is_dir()
+                    assert model_dir.stat().st_mode & 0o700
```

Comment:
If this `model.exisits()` condition is not met, I would argue that something has gone wrong. Thoughts on changing it to an `assert` instead?
```suggestion
                aseert model_dir.exists()
                assert model_dir.is_dir()
                assert model_dir.stat().st_mode & 0o700
```